File: app/src/main/java/pl/kopsoft/pczplan/models/Subject.java
package pl.kopsoft.pczplan.models;


import java.io.Serializable;

public class Subject implements Serializable {
    public String HourStart;
    public String SubjectName;
    public String Teacher;
    public SubjectType Type;
    public String Room;

    public Subject(String subjectName) {
        HourStart = "8";
        SubjectName = subjectName;
    }

    public Subject() {
        Type = SubjectType.Freiheit;
    }

    public void SetTestData() {
        HourStart = "8";
        SubjectName = "Sztuczna inteligencja";
        Teacher = "Woldan piotr";
        Type = SubjectType.Laboratory;
        Room = "s514";
    }

    public boolean IsEmpty() {
        if (SubjectName != null) {
            return SubjectName.isEmpty();
        } else {
            return true;
        }
    }
}

File: app/src/main/java/pl/kopsoft/pczplan/models/Subject.kt
package pl.kopsoft.pczplan.models


import java.io.Serializable

data class Subject(
    var hourStart: String? = null,
    var subjectName: String? = null,
    var teacher: String? = null,
    var type: SubjectType = SubjectType.Freiheit,
    var room: String? = null
) : Serializable {

    fun isEmpty(): Boolean {
        subjectName?.let {
            return it.isEmpty()
        }
        return true
    }
}
